src/app/archive_hardfork_toolbox/tests/test_convert_canonical.ml

```diff
@@ -0,0 +1,351 @@
+    ; (4, "D", Some 3, "C", 4, 3, 3, 2, "pending")
+    ; (5, "E", Some 4, "D", 5, 4, 4, 2, "pending")
+    ]
+
+  let expected_surrounded_by_pendings =
+    [ ("A", "canonical")
+    ; ("B", "orphaned")
+    ; ("C", "canonical")
+    ; ("D", "orphaned")
+    ; ("E", "orphaned")
+    ]
+
+  let all_scenarios =
+    [ ( "test_fork_on_canonical_in_the_middle"
+      , test_fork_on_canonical_in_the_middle
+      , expected_fork_on_canonical_in_the_middle
+      , "C"
+      , 2 )
+    ; ( "test_fork_on_new_network"
+      , test_fork_on_new_network
+      , expected_fork_on_new_network
+      , "C"
+      , 2 )
+    ; ( "test_fork_on_last_canonical"
+      , test_fork_on_last_canonical
+      , expected_fork_on_last_canonical
+      , "C"
+      , 2 )
+    ; ( "test_fork_on_orphaned"
+      , test_fork_on_orphaned
+      , expected_fork_on_orphaned
+      , "B"
+      , 2 )
+    ; ( "test_fork_on_pending"
+      , test_fork_on_pending
+      , expected_fork_on_pending
+      , "C"
+      , 2 )
+    ; ( "test_surrounded_by_pendings"
+      , test_surrounded_by_pendings
+      , expected_surrounded_by_pendings
+      , "C"
+      , 2 )
+    ]
+end
+
+(* Alcotest integration *)
+let test_convert_scenario conn_str
+    (name, blocks, expected, target_hash, protocol_version) () =
+  let open Deferred.Let_syntax in
+  (* Create test database *)
+  let db_name = sprintf "test_%s" name in
+  let%bind _drop =
+    TestDb.drop_database_if_exists conn_str db_name |> Deferred.ok
+  in
+  let%bind _create = TestDb.create_database conn_str db_name |> Deferred.ok in
+  let%bind _setup = TestDb.create_test_schema conn_str db_name |> Deferred.ok in
+  let%bind _insert =
+    TestDb.insert_blocks conn_str db_name blocks |> Deferred.ok
+  in
+
+  (* Run conversion *)
+  let postgres_uri = Uri.of_string (sprintf "%s/%s" conn_str db_name) in
+  let%bind _result =
+    Logic.convert_chain_to_canonical ~postgres_uri
+      ~target_block_hash:target_hash ~protocol_version ~stop_at_slot:None ()
+    |> Deferred.ok
+  in
+
+  (* Get results *)
+  let%bind actual_result = TestDb.get_all_blocks conn_str db_name in
+  let actual =
+    match actual_result with
+    | Ok blocks ->
+        blocks
+    | Error e ->
+        failwith (Error.to_string_hum e)
+  in
+
+  (* Check results match expected *)
+  let matches =
+    List.equal
+      (fun (h1, s1) (h2, s2) -> String.equal h1 h2 && String.equal s1 s2)
+      actual expected
+  in
+
+  if matches then (
+    printf "✅ Test %s passed\n" name ;
+    Deferred.return () )
+  else (
+    printf "❌ Test %s failed\n" name ;
+    printf "Expected: %s\n"
+      (List.to_string ~f:(fun (h, s) -> sprintf "%s->%s" h s) expected) ;
+    printf "Actual: %s\n"
+      (List.to_string ~f:(fun (h, s) -> sprintf "%s->%s" h s) actual) ;
+    failwith (sprintf "Test %s failed" name) )
+
+let make_test scenario conn_str =
+  Thread_safe.block_on_async_exn (fun () ->
+      test_convert_scenario conn_str scenario () )
+
+let conn_str_arg =
```

Comment:
Please consider rewrite this with Core CLI parsing. We do have a couple of cmdliner in our codebase but they're very rare. And I think their readability is bad.